malfeasance/handler.go

```diff
@@ -31,37 +29,51 @@ var (
 	errInvalidProof  = fmt.Errorf("%w: invalid proof", pubsub.ErrValidationReject)
 )
 
+type (
+	MalfeasanceHandlerV1 func(ctx context.Context, data scale.Type) (types.NodeID, []string, error)
+	MalfeasanceHandlerV2 func(ctx context.Context, data []byte) (types.NodeID, []string, error)
```

Comment:
How about removing word `Malfeasance` from the type name? It's already part of the package (it's now `malfeasance.MalfeasanceHandlerV1` instead of `malfeasance.HandlerV1`):
```suggestion
	HandlerV1 func(ctx context.Context, data scale.Type) (types.NodeID, []string, error)
	HandlerV2 func(ctx context.Context, data []byte) (types.NodeID, []string, error)
```